exp/internal/controllers/machinepool_controller_noderef.go

```diff
@@ -114,6 +110,8 @@ func (r *MachinePoolReconciler) reconcileNodeRefs(ctx context.Context, s *scope)
 	mp.Status.Deprecated.V1Beta1.ReadyReplicas = int32(nodeRefsResult.ready)
 	mp.Status.Deprecated.V1Beta1.AvailableReplicas = int32(nodeRefsResult.available)
 	mp.Status.Deprecated.V1Beta1.UnavailableReplicas = ptr.Deref(mp.Status.Replicas, 0) - mp.Status.Deprecated.V1Beta1.AvailableReplicas
+	mp.Status.ReadyReplicas = ptr.To(int32(nodeRefsResult.ready))
```

Comment:
New replica counters should be calculated based on MP Machines if the MP is using Machines. Let's please keep MachinePools entirely out of scope in this PR. The work for v1beta2 in core CAPI was not done yet for MP so I don't want to try to figure it out now when we try to finish CAPD for everything apart from MP

The MP work is tracked here separately: https://github.com/kubernetes-sigs/cluster-api/issues/12178 (including CAPD)

So tl;dr let's drop all MP and DockerMachinePool changes from this PR